musicbot/bot.py

```diff
@@ -5545,6 +5545,43 @@ async def cmd_config(
                 f"To save the change use `config save {opt.section} {opt.option}`",
                 delete_after=30,
             )
+        
+        # reset an option to default value as defined in ConfigDefaults
+        if option == "reset":
+            if not opt.editable:
+                raise exceptions.CommandError(
+                    f"Option `{opt}` is not editable. Cannot reset to default.",
+                    expire_in=30,
+                )
+
+            # Use the default value from the option object
+            default_value = opt.default
```

Comment:
As it turns out, there is also `ConfigOptionRegistry.to_ini()` which can be used to convert either the current config value or it's default value into a string suitable for use in the ini file or to pass along to `Config.update_option()`  

With that in mind, in order not to duplicate conversion logic, probably best to replace the default value parts with just this:
```py
default_value = self.config.register.to_ini(opt)
```